applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_adjoint_static_solver.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing the Kratos Library
+import KratosMultiphysics
+
+# Check that applications were imported in the main script
+KratosMultiphysics.CheckRegisteredApplications("StructuralMechanicsApplication")
+
+# Import applications
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+
+import structural_mechanics_solver
+
+def CreateSolver(model, custom_settings):
+    return StructuralMechanicsAdjointStaticSolver(model, custom_settings)
+
+class StructuralMechanicsAdjointStaticSolver(structural_mechanics_solver.MechanicalSolver):
+
+    def __init__(self, model, custom_settings):
+
+        adjoint_settings = KratosMultiphysics.Parameters("""
+        {
+            "scheme_settings" : {
+                "scheme_type": "adjoint_structural"
+            }
+        }
+        """)
+
+        self.validate_and_transfer_matching_settings(custom_settings, adjoint_settings)
+        self.scheme_settings = adjoint_settings["scheme_settings"]
+
+        self.response_function_settings = custom_settings["response_function_settings"].Clone()
+        custom_settings.RemoveValue("response_function_settings")
+        # Construct the base solver.
+        super(StructuralMechanicsAdjointStaticSolver, self).__init__(model, custom_settings)
+        self.print_on_rank_zero("::[AdjointMechanicalSolver]:: ", "Construction finished")
+
+    def AddVariables(self):
+        super(StructuralMechanicsAdjointStaticSolver, self).AddVariables()
+        self.main_model_part.AddNodalSolutionStepVariable(StructuralMechanicsApplication.ADJOINT_DISPLACEMENT)
+        if self.settings["rotation_dofs"].GetBool():
+            self.main_model_part.AddNodalSolutionStepVariable(StructuralMechanicsApplication.ADJOINT_ROTATION)
+        # TODO evaluate if these variables should be historical
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.SHAPE_SENSITIVITY)
+        self.main_model_part.AddNodalSolutionStepVariable(StructuralMechanicsApplication.POINT_LOAD_SENSITIVITY)
+        self.print_on_rank_zero("::[AdjointMechanicalSolver]:: ", "Variables ADDED")
+
+    def PrepareModelPart(self):
+        if(self.main_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]!= 3):
+            raise Exception("there are currently only 3D adjoint elements available")
+        super(StructuralMechanicsAdjointStaticSolver, self).PrepareModelPart()
+        # TODO Why does replacement need to happen after reading materials?
```

Comment:
Probably bcs the Properties are not exchanged in the ReplacementProcess?

Reply:
I found out why:
 If the primal element is replaced with the adjoint element, the read materials process sets the property at the adjoint element, but the primal element does not notice this. Because the SetProperties() function at the element base class is not virtual, i can not overwrite it in my adjoint element, so i did not find a way to forward the property to the primal element.

Eventually it could be done in the Initialize() of the adjoint element.